core/connection.js

```diff
@@ -208,6 +219,60 @@ Blockly.Connection.prototype.checkConnection_ = function(target) {
 };
 
 /**
+ * Check if the two connections can be dragged to connect to each other.
+ * @param {Blockly.Connection} candidate A nearby connection to check.
+ * @param {number} maxRadius The maximum radius allowed for connections.
+ * @return {boolean} True if the connection is allowed, false otherwise.
+ */
+Blockly.Connection.prototype.isConnectionAllowed = function(candidate,
+   maxRadius) {
+  if (this.distanceFrom(candidate) > maxRadius) {
+      return false;
+  }
+
+  // Type checking
+  var canConnect = this.canConnectWithReason_(candidate);
+  if (canConnect != Blockly.Connection.CAN_CONNECT
+          && canConnect != Blockly.Connection.REASON_MUST_DISCONNECT) {
+      return false;
+  }
+
+  // Don't offer to connect an already connected left (male) value plug to
+  // an available right (female) value plug.  Don't offer to connect the
+  // bottom of a statement block to one that's already connected.
+  if (candidate.type == Blockly.OUTPUT_VALUE
+      || candidate.type == Blockly.PREVIOUS_STATEMENT) {
```

Comment:
|| belongs at end of previous line in JS (thanks to the bad idea than semicolons are optional).